testhelper/httptest/httptest.go

```diff
@@ -0,0 +1,47 @@
+package httptest
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	nethttptest "net/http/httptest"
+)
+
+// NewServer starts a new httptest server that listens on all interfaces, contrary to the standard net/httptest.Server that listens only on localhost.
+// This is useful when you want to access the test http server from within a docker container.
+func NewServer(handler http.Handler) *Server {
+	ts := newUnStartedServer(handler)
+	ts.Start()
+	return ts
+}
+
+// Simple net/httptest.Server wrapper
+type Server struct {
+	*nethttptest.Server
+}
+
+func (s *Server) Start() {
```

Comment:
Its required to override Start method of net/httptest.Server

Reply:
Calling the `func (s *Server) Start()` method on `NewServer` by the customer will result in `panic: Server already started`. Can u check this once?

Also, no need to overwrite the Start since the customer can't prepare the `kithttptest.Server `struct because newUnStartedServer is not exported, we don't want it to be overridden as well.

Reply:
Yeah, not needed to override as we are calling this internally